pkgs/top-level/static.nix

```diff
@@ -43,6 +43,10 @@ self: super: let
     # ++ optional (super.stdenv.hostPlatform.libc == "glibc") ((flip overrideInStdenv) [ self.stdenv.glibc.static ])
   ;
 
+  exposeStatic = super: super // {
+    targetPlatform = super.targetPlatform // { isStatic = true; };
```

Comment:
So we shouldn't be overriding `*Platform` in overlays, for a number of obscure reasons.

Instead, we should change the definition of `pkgsStatic` to include `isStatic = true;` in the `crossSystem` argument to Nixpkgs.

That, and the `pkgs/stdenv/cross/default.nix` line I already pointed out above, and are the only places `static.nix` is used. The fancier defaulting suggestion I made ensures that latter is already in sync with the use of the overlay, so changing `pkgsCross` like so is the only other fixup needed.